pkg/catalogsourceconfig/registry.go

```diff
@@ -235,26 +243,28 @@ func (r *registry) getLabel() map[string]string {
 	return map[string]string{"marketplace.catalogSourceConfig": r.csc.GetName()}
 }
 
-// getOperatorSources returns a concatanated string of namespaced OperatorSources
-// and an array of OperatorSource names where the packages in the
-// CatalogSourceConfig are present.
-func (r *registry) getOperatorSources() (string, []string) {
-	var opsrcString string
-	var opsrcList []string
+// getAppRegistryCmdLineOptions returns a group of "--registry=" command line
+// option(s) required for operator-registry. If one of the packages is from a
+// private repository secretIsPresent will be true.
+func (r *registry) getAppRegistryCmdLineOptions() (appRegistryOptions string, secretIsPresent bool) {
 	for _, packageID := range r.csc.Spec.GetPackageIDs() {
 		opsrcMeta, err := r.reader.Read(packageID)
 		if err != nil {
 			r.log.Errorf("Error %v reading package %s", err, packageID)
 			continue
 		}
-		opsrcNamespacedName := opsrcMeta.Namespace + "/" + opsrcMeta.Name
-		if !strings.Contains(opsrcString, opsrcNamespacedName) {
-			opsrcString += opsrcNamespacedName + ","
-			opsrcList = append(opsrcList, opsrcMeta.Name)
+		//--registry="https://quay.io/cnr|community-operators" --regisry="https://quay.io/cnr|custom-operators|mynamespace/mysecret"
+		appRegistry := "--registry=" + opsrcMeta.Endpoint + "|" + opsrcMeta.RegistryNamespace
```

Comment:
looks like the pipe delimited soure is not enclosed in double quotes `--registry="https://quay.io/cnr|community-operators"`